### PR TITLE
Upgrade SQLAlchemy to version 2, update CI

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,17 +26,21 @@ help:
...
+
+.PHONY: poetry
+poetry:
+	poetry install --with=dev --extras=psycopg2-binary
 
-.PHONY: flake8
-flake8: .build/requirements-dev.timestamp
-	$(VENV_BIN)/flake8 --exclude=node_modules c2cgeoform
+.PHONY: prospector
+prospector: poetry
```
Comment (arnaud-morvan) [inline Makefile:39]: Why not running prospector using pre-commit ?

**Review by arnaud-morvan — COMMENTED** [review]
Looks good.

I tested locally so I added 2 commits to update dev docs.

For contributing in README, we should put a link to developer documentation.

I still get a big diff on package.lock file after build, maybe related to the node version used.

File: Makefile (hunk 1 above)
Comment (sbrunner) [reply Makefile:39]: For tow reason:
- Proespector run is too slow: pre-comit run in less than 10s. (8.6s. on my host on all files, on second run) and prospector in 1m45s (same condition).
- Prospector needs the project dependencies

**Comment by sbrunner** [comment]
To fix the lock file, I updated to node 16 (and use nvmrc do don't have the version in two places) Any objection?

**Comment by sbrunner** [comment]
Note that version under 18 are no more supported...
https://endoflife.date/nodejs

**Review by arnaud-morvan — APPROVED** [review]
LGTM, still an error with snyk, but cannot see it. I requested access to the organization.

**Pull request merged**